### ISSUE TITLE
[Bug] Data View Namespaces Field Null during import

### ISSUE THREAD

**Issue by btrieger** [elastic/terraform-provider-elasticstack#671]
**Describe the bug**
When importing a data view to the terraform state file the namespaces field is set to null despite it being shared with all spaces and the api showing namespaces: ["*","default"]

**To Reproduce**
1. TF configuration used '...'
2. TF operations to execute to get the error '...' [e.g `terraform plan`,`terraform apply`, `terraform destroy`]
3. See the error in the output '...'
1. Import a data view
2. terraform import
3. View tfstate file
**Expected behavior**
I expect the namespaces field to match what the kibana api returns for data views. 

**Debug output**
Run `terraform` command with `TF_LOG=trace` and provide extended information on TF operations. **Please ensure you redact any base64 encoded credentials from your output**. 
eg
```
[DEBUG] provider.terraform-provider-elasticstack_v0.11.0: Authorization: Basic xxxxx==
```
[debug.txt](https://github.com/user-attachments/files/15972739/debug.txt)

**Screenshots**
![image](https://github.com/elastic/terraform-provider-elasticstack/assets/56197436/71b0861c-bf69-4736-8dc1-40f92eac0f5e)
<img width="685" alt="image" src="https://github.com/elastic/terraform-provider-elasticstack/assets/56197436/d685afc9-c04e-4e53-a3d5-8f8afa80b35b">

**Versions (please complete the following information):**
 - OS: [e.g. Linux]
 - Terraform Version [e.g. 1.0.0] 1.2.2
 - Provider version [e.g. v0.1.0]  v0.11.4
 - Elasticsearch Version [e.g. 7.16.0] 8.7.0

**Additional context**

**Comment by btrieger** [comment]
[trace.txt](https://github.com/user-attachments/files/15972780/debug.txt)
Trace level attached initial was debug level

**Comment by AnanthRavi** [comment]
any updates on this ticket? Running into the same issue where the namespaces field is ignored while importing into the state.

**Comment by AnanthRavi** [comment]
@daemitus would the PR you recently merged https://github.com/elastic/terraform-provider-elasticstack/pull/1011 fix this issue too?

**Comment by daemitus** [comment]
Looks like there is still one edge-case where this is still a problem, however it looks mostly solved.

When given the following during an import:
- 1: `namespaces = null` works great.
- 2: `namespaces = ["default", "something else"]` works great (provided the order is correct)
- 3:`namespaces = ["default"]` does not work. This runs afoul of a semantic check that is looking for # 1. We'd need to pass in what's in the plan too, rather than just comparing the current state against the API response.

**Comment by daemitus** [comment]
fwiw, I give all my dataviews explicit IDs, so if I end up with an import problem I just let terraform recreate it. At least the dashboards or other places it gets used won't break.

**Comment by AnanthRavi** [comment]
@daemitus should I create a ticket for edge case 3? We have loads of data views that are present only in the default namespace and we might have issues if we force re-creating the data views? We have had cases where some of the managed dashboards break  when the data views are re-created. Let me know if there is a better way of handling this or if I should file an issue?

**Comment by daemitus** [comment]
I might have an idea to get it working, let me carve out a few hours in the
next couple days and see what I can do.

**Comment by daemitus** [comment]
Nah, I don't think theres a way around this with regards to importing dataviews. iirc, this same sort of issue happens when importing indexes or index templates. If you have settings specified, the import will try to remove them if they aren't specified in the tf files. It is just a larger problem here, because namespace drift triggers a replace. If you routinely need to do imports, I'd recommend always including your default namespace, or whatever is in `space_id`  in the `namespaces` field, which should solve the problem.

**Comment by AnanthRavi** [comment]
@daemitus lets say we have data views that we would not like to be shown to the default space. Is there a workaround? The main concern here is that replacing the data view might break our dashboards. Is this still the case as the id of the data view might break existing dashboards?

**Comment by daemitus** [comment]
`[space_id](https://registry.terraform.io/providers/elastic/elasticstack/latest/docs/resources/kibana_data_view#space_id-1) (String) An identifier for the space. If space_id is not provided, the default space is used.`

If the space_id parameter is null/unset, it becomes `default`. Each dataview basically needs a home, so if you only want it to be in prod and dev, but not default, you would need to set the space_id to one of those, and use the namespaces field to propagate it to the other. In the first case, if "default" is the space_id, but isn't in namespaces, the next time terraform is run it will 404 and then throw an error when it tries to recreate something that already exists.

If you set the `dataview.id` field, you can freely recreate the views all day long. The references in the dashboard are just the id. So if you aren't, you could take the autogenerated ones, plug it into the terraform, then let it do whatever its going to do.
